client/game.cpp

```diff
@@ -1,165 +1,154 @@
+	});
+}
 
-	void Game::Startup() {
-		this->rs.Startup();
-		const unsigned int window_width = this->config_script->environment.get_or("window_width", WINDOW_WIDTH);
-		const unsigned int window_height = this->config_script->environment.get_or("window_height", WINDOW_HEIGHT);
-		this->rs.SetViewportSize(window_width, window_height);
-	}
+void Game::Startup() {
+	this->rs.Startup();
+	const unsigned int window_width = this->config_script->environment.get_or("window_width", WINDOW_WIDTH);
+	const unsigned int window_height = this->config_script->environment.get_or("window_height", WINDOW_HEIGHT);
+	this->rs.SetViewportSize(window_width, window_height);
+}
 
-	Game::~Game() {
-		this->config_script.reset();
-		this->ss.Stop();
-		this->sound_thread.join();
+Game::~Game() {
+	this->config_script.reset();
+	this->ss.Stop();
+	this->sound_thread.join();
 
-		this->server_connection.Disconnect();
-		this->server_connection.Stop();
-		if (this->asio_thread) {
-			this->asio_thread->join();
-			delete this->asio_thread;
-		}
-		if (this->sync_thread) {
-			this->sync_thread->join();
-			delete this->sync_thread;
-		}
+	this->server_connection.Disconnect();
+	this->server_connection.Stop();
+	if (this->asio_thread) {
+		this->asio_thread->join();
+		delete this->asio_thread;
+	}
+	if (this->sync_thread) {
+		this->sync_thread->join();
+		delete this->sync_thread;
 	}
+}
 
-	void Game::UpdateVComputerScreenTextures() {
-		tda::TDAScreen screen;
-		static PixelBuffer local_pbuffer(320, 240, 8, ImageColorMode::COLOR_RGBA);
-
-		for (auto computer_itr = ComputerComponentMap::Begin(); computer_itr != ComputerComponentMap::End(); ++computer_itr) {
-			auto comp = *computer_itr;
-			std::shared_ptr<ComputerScreen> comp_screen = std::static_pointer_cast<ComputerScreen>(comp.second->devices[5]);
-			std::static_pointer_cast<tda::TDADev>(comp_screen->device)->DumpScreen(screen);
-			std::lock_guard<std::mutex> lock(local_pbuffer.GetWritelock());
-			tda::TDAtoRGBATexture(screen, (std::uint32_t*)local_pbuffer.GetBlockBase());
-			if (comp_screen->texture) {
-				comp_screen->texture->Load(local_pbuffer);
-			}
-			else {
-				Entity screen_entity(comp.first);
-				if (screen_entity.Has<Renderable>()) {
-					const auto* ren = screen_entity.Get<Renderable>();
-					if (ren->buffer) {
-						if (ren->buffer->GetVertexGroupCount() > 0) {
-							auto texture_instance = std::make_shared<TextureObject>(local_pbuffer);
-							auto material = Material::Create(std::to_string(comp.first) + "_screen");
-							material->AddTexture(texture_instance);
-							ren->buffer->GetVertexGroup(0)->material = material;
-							comp_screen->texture = texture_instance;
-						}
+void Game::UpdateVComputerScreenTextures() {
+	tda::TDAScreen screen;
+	static PixelBuffer local_pbuffer(320, 240, 8, ImageColorMode::COLOR_RGBA);
+
+	for (auto computer_itr = ComputerComponentMap::Begin(); computer_itr != ComputerComponentMap::End();
```

Comment:
This could be cleaned up with a range based for loop or similar.

Reply:
Correction because of how multiton's interface is this cannot work, so I just shortened the variable's name to itr